apps/dashboard/app/javascript/packs/batch_connect_sessions.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const pollers = [];
+
+class Timer {
+  constructor(callback, delay){
+    this.delay = delay;
+    this.remaining = delay;
+    this.active = true;
+    this.callback = callback;
+
+    this.resume();
+  }
+
+  resume() {
+    if(!this.active) { return; }
+
+    this.start = new Date();
+    this.clearTO();
+    this.timerId = setTimeout(this.callback, this.remaining);
+  }
+
+  restart() {
+    if(!this.active) { return; }
+
+    this.remaining = this.delay;
+    resume();
+  }
+
+  pause() {
+    if(!this.active) { return; }
+
+    this.clearTO();
+    this.remaining -= new Date() - this.start;
+  }
+
+  stop(){
+    if(!this.active) { return; }
+    this.clearTO();
+    this.active = false;
+  }
+
+  clearTO(){
+    if(this.timerId !== undefined) {
+      clearTimeout(this.timerId);
+    }
+  }
+
+}
+
+class Poller {
+  constructor(url, delay) {
+    this.url = url;
+    this.delay = delay;
+    this.poll();
+  }
+
+  poll(){
+    this.timer = new Timer(this.request.bind(this), this.delay);
+  }
+
+  pause() {
+    this.timer.pause();
+  }
+
+  resume() {
+    this.timer.resume();
+  }
+
+  request(){
+    const that = this;
+    $.getScript(this.url)
+      .done((_script, _textStatus, _jqxhr) => {
+        return;
+      }).fail((_jqxhr, textStatus, errorThrown) => {
+        if(textStatus) { console.log(`Failed to get session data. Server returned '${textStatus}'`); }
+        if(errorThrown) { 
+          console.log("Failed to get session data because of error.");
+          console.log(errorThrown);
+        }
+        return;
+      }).always(() => {
+        that.poll();
+        return;
+      });
+  }
+}
+
+function makePollers(){  
+  const obj = $('#batch_connect_sessions');
+  if(!obj) return;
```

Comment:
Originally this was 

```coffeescript
$('[data-toggle="poll"]').each
    url   = $(this).data('url')
    delay = $(this).data('delay')
```

but I was having trouble here with `$(this)` not returning correctly. So instead let's just query for 1 single ID because really, that's what we're after here 1 single div that has the polling information, not many.